spec/acceptance/exec_pwsh_spec.rb

```diff
@@ -456,55 +402,76 @@ def platform_string(host, windows, posix)
-      }
-      MANIFEST
-      describe file('c:/temp/services.csv') do
-        # Due to https://tickets.puppetlabs.com/browse/QA-3461 each host must be done one at a time
-        windows6_agents.each { |hut| execute_manifest_on(hut, p2, :catch_failures => true) }
+    let(:manifest) { <<-MANIFEST
+    file{'#{external_script}':
+      content => '#{File.open(File.join(File.dirname(__FILE__), external_fixture)).read()}'
+    }
+    exec{"TestPowershellPS1":
+      command   => '#{external_script}',
+      provider  => pwsh,
+      require   => File['#{external_script}']
+    }
+    MANIFEST
+    }
+
+    win_file = 'c:/temp/commands.csv'
+    posix_file = '/tmp/commands.csv'
+
+    powershell6_agents.each do |node|
+      describe file(windows_platform?(node) ? win_file : posix_file), :node => node do
+        let(:external_script) { windows_platform?(node) ? 'c:/external-script.ps1' : '/external-script.ps1' }
+        let(:external_fixture) { "files/get-command-#{platform_string(node, 'win', 'posix')}.ps1" }
+
+        it 'should apply the manifest' do
+          #require 'pry'; binding.pry
+          execute_manifest_on(node, manifest, :catch_failures => true)
+        end
+
         it { should be_file }
-        its(:content) { should match /puppet/ }
+        its(:content) { should match /Get-Command/ }
       end
     end
   end
 
   describe 'passing parameters to the ps1 file' do
-    context 'on Windows platforms', :if => windows6_agents.count > 0 do
-      outfile = 'C:/temp/svchostprocess.txt'
-      processName = 'svchost'
-      pp = <<-MANIFEST
-        $process = '#{processName}'
-        $outFile = '#{outfile}'
-      file{'c:/param_script.ps1':
-        content => '#{File.open(File.join(File.dirname(__FILE__), 'files/param_script.ps1')).read()}'
+    let(:manifest) { <<-MANIFEST
+      $commandName = '#{commandName}'
+      $outFile = '#{outfile}'
+
+      file{'#{external_script}':
+        content => '#{File.open(File.join(File.dirname(__FILE__), 'files/param_script-posix.ps1')).read()}'
       }
       exec{'run this with param':
         provider => pwsh,
-        command	 => "c:/param_script.ps1 -ProcessName '$process' -FileOut '$outFile'",
-        require  => File['c:/param_script.ps1'],
-      }
-      MANIFEST
-      describe file(outfile) do
-        # Due to https://tickets.puppetlabs.com/browse/QA-3461 each host must be done one at a time
-        windows6_agents.each { |hut| execute_manifest_on(hut, pp, :catch_failures => true) }
+        command	 => "#{external_script} -CommandName '$commandName' -FileOut '$outFile'",
+        require  => File['#{external_script}'],
+    }
+    MANIFEST
+    }
+
+    win_file = 'c:/temp/params.csv'
+    posix_file = '/tmp/params.csv'
+
+    powershell6_agents.each do |node|
+      describe file(windows_platform?(node) ? win_file : posix_file), :node => node do
+        let(:external_script) { windows_platform?(node) ? 'C:\\param_script.ps1' : '/param_script.ps1' }
```

Comment:
Did you try using just `/param_script.ps1` across all platforms?

Reply:
No I didn't. but  I given I had to do it differently per platform anyway, it seemed the right thing to do